week3/homework/src/index.js

```diff
@@ -1,3 +1,112 @@
 'use strict';
 
 // Write the homework code in this file
+
+const express = require('express');
+const uuidv4 = require('uuid/v4');
+
+const app = express();
+
+const PORT = 3000;
+
+const {writeFile, readFile} = require('./todos.js');
+// const writeFile = todos.writeFile;
+// const readFile = todos.readFile;
+
+app.use(express.json());
+// ===============================readAllToDos========================================
+app.get('/todos', (req, res) => {
+    readFile()
+    .then(data => {
+        res.json(data);
+        res.end();
+    });
+});
+// ===============================giveIdForEachToDo========================================
+app.post('/todos', (req, res) => {
+    readFile()
+        .then(todos => {
+            const todoToAdd = req.body.todo;
+            todoToAdd.id = uuidv4();
+            todos.push(todoToAdd);
+            writeFile(todos)
+                .then(() => {
+                    res.send(`"${req.body.todo.description}" added to the list of toDos.`);
+                    res.end();
+                });
+        });
+});
+// ===============================deleteSelectedIdOfToDo========================================
+app.delete('/todos/:id', (req, res) => {
+    const removedId = req.params.id;
+    readFile()
+        .then(todos => {
+            const newTodos = todos.filter(todo => todo.id !== removedId);
+            writeFile(newTodos)
+                .then(() => {
+                    res.send(`"${removedId}" "removed" from the list of toDos.`);
+                    res.end();
+                });
+        });
+});
+// ===============================deleteAllTodoList========================================
+app.delete('/todos', (req, res) => {
+    readFile()
```

Comment:
What is the point of reading all to-dos first if you're not going to do anything with the result before overwriting them?